bonelab/cli/PointPicker_implement.py

```diff
@@ -0,0 +1,327 @@
+
+"""
+Author: Matt Daalder & Bryce Besler
+Last edited: Feb 23, 2020
+
+Pick points on a nifti, dicom, or aim file.
+This code is optimized for bone.
+
+Takes as input the medical image to read and the directory in which to write the
+picked points plain text file.
+
+Example command line/terminal input:
+    >>>python PointPicker_implement.py \input\path\image.nii \output\directory\path\
+
+Press 'p' to place a point at the location of your cursor.
+Press 'z' to remove a point at the location of your cursor.
+Press 't' to write a plaintext file containing the coordintaes of the points picked.
+
+This code takes the medical image, applies a gaussian filter, segments the filtered 
+image based on bone CT HU values then renders the surface using marching cubes.
+"""
+
+# Imports
+import argparse
+import os
+import vtk
+
+from bonelab.util.echo_arguments import echo_arguments
+from bonelab.io.points import write_points
+from bonelab.io.vtk_helpers import get_vtk_reader
+
+# =============================================================================
+# Takes as argument input file of type nifti, dicom, or aim
+# Reads the files, applies a gaussian filter, segments based on bone HU threshold values
+# then renders the surface using marching cubes algorithm.
+# =============================================================================
+def PointPicker(input_filename, output_dir, overwrite=False):
+
+    # Check if output directory exists, if not, exit
+    if not os.path.isdir(output_dir):
+        os.sys.exit('[ERROR] Cannot find output directory \"{}\" in which to write picked points .txt file'.format(output_dir))
+        
+    # Read input
+    if not os.path.isfile(input_filename):
+        os.sys.exit('[ERROR] Cannot find file \"{}\"'.format(input_filename))
+
+    reader = get_vtk_reader(input_filename)
+    if reader is None:
+        os.sys.exit('[ERROR] Cannot find reader for file \"{}\"'.format(input_filename))
+
+    print('Reading input image ' + input_filename)
+    reader.SetFileName(input_filename)
+    reader.Update()
+    
+    # apply filter(s) to images to smooth it out
+    filteredImg = vtk.vtkImageGaussianSmooth()
+    filteredImg.SetInputConnection(reader.GetOutputPort())
+    filteredImg.SetStandardDeviation(1)
+    filteredImg.SetRadiusFactors(1,1,1)
+    filteredImg.SetDimensionality(3)
+    filteredImg.Update()
+    
+    """ render and display the image """
+       
+    #set threshold values for segmentation
+    loThresh = 200
+    hiThresh = 3000
```

Comment:
Both should be parameters specifiably at the command line or through a graphical interface.